### ISSUE TITLE
Memory leak in RNG _sources?

### ISSUE THREAD

**Issue by talex5** [mirage/mirage-crypto#94]
According to `perf`, ocluster-scheduler is spending a lot of time in `List.length`:
```
-   57.92%  ocluster-schedu  ocluster-scheduler  [.] camlStdlib__list__length_aux_83
     camlStdlib__list__length_aux_83
     camlMirage_crypto_rng__Entropy__register_source_143
     camlMirage_crypto_rng__Entropy__cpu_rng_596
     camlMirage_crypto_rng_lwt__one_151
     camlLwt__callback_1288
     camlLwt__iter_callback_list_873
     camlLwt__run_in_resolution_loop_926
     camlLwt__resolve_946
     camlLwt__wakeup_general_983
     caml_start_program
```

I don't really understand this code, but it looks like `initialize` calls `rdrand_task` https://github.com/mirage/mirage-crypto/blob/ab6c5f323d7a848a85ecaa091c380076f3c11d1a/rng/lwt/mirage_crypto_rng_lwt.ml#L61

That then calls [`Entropy.cpu_rng`](https://github.com/mirage/mirage-crypto/blob/ab6c5f323d7a848a85ecaa091c380076f3c11d1a/rng/entropy.ml#L139) once per second. Each time, that calls `register_source`, which gets the length of `_sources` and then appends to it. As far as I can see, nothing is ever removed from this list.

**Comment by hannesm** [comment]
Thanks for your report, this is indeed a leak. I prepared https://github.com/mirage/mirage-crypto/pull/95 which should fix the issue you are encountering. Would you mind to test it and report here? A bugfix release of mirage-crypto is scheduled soon.

**Comment by talex5** [comment]
I've deployed the fix on the live cluster scheduler. If your ocaml-ci and opam-repo-ci jobs keep working, then the patch is good :-)

**Comment by talex5** [comment]
Here's a graph of CPU usage *without* the patch. It will take a few days to see the effect of the fix:

![leak](https://user-images.githubusercontent.com/554131/103346672-9b851880-4a8c-11eb-8343-c95b5598fc69.png)

**Comment by hannesm** [comment]
thanks again. the fix has been integrated into 0.8.8, now PRed to opam-repository https://github.com/ocaml/opam-repository/pull/17917
